Service.cs

```diff
@@ -0,0 +1,151 @@
+            bool isCardNominalCorrect = int.TryParse(Convert.ToString(combination[nominalPosition]), out int enteredNominal);
+            if (!isCardNominalCorrect)
+            {
+                ErrorMessage = "\nInvalid value of card nominal!";
+                isDataCorrect = false;
+            }
+            else
+            {
+                if (enteredNominal > 7 || enteredNominal < 1)
+                {
+                    ErrorMessage = "\nCard nominal should be in the range [1, 7]!";
+                    isDataCorrect = false;
+                }
+                else
+                {
+                    if (!Enum.TryParse(Convert.ToString(combination[colorPosition]), out Colors.SevenColors color))
+                    {
+                        ErrorMessage = "Invalid value of color";
+                        isDataCorrect = false;
+                    }
+                    CardNominal = enteredNominal;
+                    CardColor = Colors.GetDisplayName(color);
+                }
+            }
+            return isDataCorrect;
+        }
+        public static bool AreAnyIdenticalCards(List<Card> cards1, List<Card> cards2)
+        {
+            for (int i = 0; i < cards1.Count; i++)
+            {
+                for (int j = 0; j < cards2.Count; j++)
+                {
+                    if (cards1[i].Equals(cards2[j]))
+                    {
+                        ErrorMessage = "There can be no identical cards";
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+        public static int GetIndexOfHigherCard(string color)
+        {
+            int ind;
+            if (color == Colors.GetDisplayName(Colors.SevenColors.R))
+            {
+                ind = (int)Colors.SevenColors.R;
+            }
+            else if (color == Colors.GetDisplayName(Colors.SevenColors.O))
+            {
+                ind = (int)Colors.SevenColors.O;
+            }
+            else if (color == Colors.GetDisplayName(Colors.SevenColors.Y))
+            {
+                ind = (int)Colors.SevenColors.Y;
+            }
+            else if (color == Colors.GetDisplayName(Colors.SevenColors.G))
+            {
+                ind = (int)Colors.SevenColors.G;
+            }
+            else if (color == Colors.GetDisplayName(Colors.SevenColors.C))
+            {
+                ind = (int)Colors.SevenColors.C;
+            }
+            else if (color == Colors.GetDisplayName(Colors.SevenColors.B))
+            {
+                ind = (int)Colors.SevenColors.B;
+            }
+            else
+            {
+                ind = (int)Colors.SevenColors.P;
+            }
+            return ind;
+        }
+        public static void FindHighCard(int combLen, List<Card> cards)
+        {
+            int higherCardNominal = cards[0].Nominal, higherCardNumber = 0;
+            for (int i = 1; i < combLen; i++)
```

Comment:
Why do you need `comLen` variable?  Do we have any other ways to get the length of the `cards` collection?